lib/core/capabilities/VoicePackManagementCapability.js

```diff
@@ -0,0 +1,51 @@
+const NotImplementedError = require("../NotImplementedError");
+const Capability = require("./Capability");
+
+class VoicePackManagementCapability extends Capability {
+    /**
+     * Returns the current applied voice pack language.
+     *
+     * @abstract
+     * @returns {Promise<string>}
+     */
+    async getCurrentVoiceLanguage() {
+        throw new NotImplementedError();
+    }
+
+    /**
+     * This method should instruct the vacuum to download a voice pack from `presignedUrl`.
+     * The actual specifications of what exactly is hosted behind presignedUrl depend on the specific vacuum model.
+     * The same goes for the hash, the user should provide a hash or signature as expected by the vacuum.
+     *
+     * @abstract
+     * @param {object} options
+     * @param {string} options.presignedUrl
+     * @param {string} [options.language]
+     * @param {string} [options.hash]
+     * @returns {Promise<void>}
+     */
+    async downloadVoicePack(options) {
+        throw new NotImplementedError();
+    }
+
+    /**
+     * This method should return the progress of the current voice pack operation.
+     * The progress should be returned as a percentage, null if the operation has completed or no operation in progress.
+     *
+     * @returns {Promise<[number]>}
+     */
+    async getVoicePackOperationProgress() {
```

Comment:
Since `null` is something best avoided, I'd propose having a `ValetudoVoicePackOperationStatus` entity in `entities/core` with a `type` property that can be

- Idle
- Downloading
- Installing

and a `progress` property that will be the progress as a number in percent 0-100.

I guess idle should be 100% progress?

Reply:
Addressed in cc48ab3.